subsys/shell/shell.c

```diff
@@ -1690,6 +1690,13 @@ void shell_set_bypass(const struct shell *sh, shell_bypass_cb_t bypass)
 	sh->ctx->bypass = bypass;
 }
 
+bool shell_ready(const struct shell *sh)
+{
+	__ASSERT_NO_MSG(sh);
+
+	return state_get(sh) ==	SHELL_STATE_ACTIVE;
```

Comment:
wondering if this could be generalised as `shell_get_state`, since `SHELL_STATE_ACTIVE` and its friends are part of the public API header.

Reply:
Enum: `enum shell_state` is not part of the public API. It has a comment `internal`.

Reply:
Fair enough.